docs/API.md

```diff
@@ -278,14 +305,14 @@ Blocking and Non-blocking Mode
 SRT functions can also work in blocking and non-blocking mode, for which
 there are two separate options for sending and receiving: `SRTO_SNDSYN` and
 `SRTO_RCVSYN`. When blocking mode is used, a function will not exit until
-the availability condition is satisfied; in non-blocking mode the function 
+the availability condition is satisfied. In non-blocking mode the function 
 always exits immediately, and in case of lack of resource availability, it
-returns an error with appropriate code. The use of non-blocking mode usually
+returns an error with an appropriate code. The use of non-blocking mode usually
 requires using some polling mechanism, which in SRT is **EPoll**.
 
 Note also that the blocking and non-blocking modes apply not only for sending
-and receiving. For example, SNDSYN defines blocking for `srt_connect` and
-RCVSYN defines blocking for `srt_accept`. The SNDSYN also makes `srt_close`
+and receiving. For example, `SNDSYN` defines blocking for `srt_connect` and
+`RCVSYN` defines blocking for `srt_accept`. The `SNDSYN` also makes `srt_close`
```

Comment:
That should be updated as this is completely wrong.

The blocking for `srt_connect` is set by setting `SRTO_RCVSYN` to false (surprisingly, as the connection-ready state is `SRT_EPOLL_OUT`). The blocking for `srt_accept` is, also surprisingly, set by `SRTO_RCVSYN` on the listener socket from which the accepting is done, although the readiness for acceptance is marked by `SRT_EPOLL_IN` event.

Since the second sentence, I'd add rather something like this:

To make `srt_connect` running asynchronously, you should set the `SRTO_RCVSYN` to false on the socket used for connecting (the `SRT_EPOLL_OUT` readiness should be checked here). To make the `srt_accept` running asynchronously, you should set `SRTO_RCVSYN` to false on the listener socket (the `SRT_EPOLL_IN` readiness should be checked here). The `SRTO_SNDSYN` set to true on the socket makes also `srt_close` exit only after the sending buffer is completely empty.

Reply:
(Updated the comment above - I messed up a bit with explanations, but the text in the last paragraph is ok.)

Reply:
First of all, to make you imagine how it works: The `SRTO_SNDSYN` and `SRTO_RCVSYN` options are set to `true` by default, and this means **blocking** mode. So you should set appropriate option to `false` in order to have **non-blocking** mode. So, among things mentioned here:
* `SRTO_RCVSYN` set to `false` makes `srt_connect` run asynchronously (non-blocking)
* IDEM (on the listener socket), makes `srt_accept` run asynchronously (non-blocking)
* When `SRTO_SNDSYN` is true (default!) then also `srt_close` will block until the sender buffer is empty. When this is set to false, the `srt_close` function exits immediately, but sending will continue in the background.

Reply:
Blocking and Non-blocking Mode
==============================

SRT functions can also work in blocking and non-blocking modes, for which
there are two separate options for sending and receiving: `SRTO_SNDSYN` and
`SRTO_RCVSYN` (both are set to true by default, corresponding to blocking mode). 
When blocking mode is used, a function will not exit until the availability 
condition is satisfied. In non-blocking mode the function always exits immediately, 
and if enough resources are not available it returns an error with an appropriate 
code. The use of non-blocking mode usually requires using some polling mechanism, 
which in SRT is **EPoll**.

Note also that blocking and non-blocking modes apply not only to sending and 
receiving. For example, to make `srt_connect` run asynchronously (non-blocking mode), 
set `SRTO_RCVSYN` to false on the socket used for connecting (the `SRT_EPOLL_OUT` 
readiness should be checked here). To make `srt_accept` run asynchronously (non-blocking 
mode), set `SRTO_RCVSYN` to false on the listener socket (the `SRT_EPOLL_IN` readiness 
should be checked here). When `SRTO_SNDSYN` is true (default) then also `srt_close` 
will block until the sender buffer is empty. When `SRTO_SNDSYN` is set to false, the 
`srt_close` function exits immediately, but sending will continue in the background.

Reply:
Perfect.